streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorNode.java

```diff
@@ -131,13 +155,10 @@ public void process(final K key, final V value) {
                     valueClass),
                 e);
         }
-        nodeMetrics.nodeProcessTimeSensor.record(time.nanoseconds() - startNs);
     }
 
     public void punctuate(final long timestamp, final Punctuator punctuator) {
-        final long startNs = time.nanoseconds();
-        punctuator.punctuate(timestamp);
-        nodeMetrics.nodePunctuateTimeSensor.record(time.nanoseconds() - startNs);
+        maybeMeasureLatency(() -> punctuator.punctuate(timestamp), time, punctuateSensor);
```

Comment:
See my other comment: in `maybeMeasureLatency`, should we also check if the sensor is an empty sensor as well?

Reply:
How about this: in `Sensor` we expose a public function `hasStats` which checks if its `stats` is empty or not. If it is empty then it means no `sensor.add` has ever triggered yet (e.g. all the `emptySensor` would be the case), in which case we can skip measuring at all. By doing this we can also remove the extra check on the caller for the config value.

Reply:
Good idea! I opened PR ##7692 for the new method on `Sensor`.